src/opm/output/eclipse/AggregateMSWData.cpp

```diff
@@ -610,6 +626,72 @@ namespace {
                 usys.from_si(M::length, sicd.length());
         }
 
+        template <class RSegArray>
+        void assignAICDCharacteristics(const ::Opm::Segment&    segment,
+                                            const ::Opm::UnitSystem& usys,
+                                            const int                baseIndex,
+                                            RSegArray&               rSeg)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::VectorItems::RSeg::index;
+            using M  = ::Opm::UnitSystem::measure;
+
+            const auto& aicd = segment.autoICD();
+
+            rSeg[baseIndex + Ix::DeviceBaseStrength] =
+                usys.from_si(M::aicd_strength, aicd.strength());
+
+            rSeg[baseIndex + Ix::ScalingFactor] = ((aicd.methodFlowScaling() == 1) ||
```

Comment:
Please simplify - this is mouthful to read.

Reply:
I have added some explanatory comments for the logic test used to explain, is that OK?

Reply:
I did not really find a simple way myself either :-(